src/bin/client.rs

```diff
@@ -0,0 +1,217 @@
+//! Simple client program
+//! Establishes connection to server and sends login information
+
+extern crate uosql;
+extern crate bincode;
+extern crate bufstream;
+extern crate byteorder;
+
+use std::io::{Write, Read};
+use std::io;
+use std::io::stdout;
+use std::net::TcpStream;
+use bincode::rustc_serialize::{decode_from, encode_into};
+use uosql::net::Cnv;
+use uosql::net::Greeting;
+use uosql::net::Login;
+use uosql::net::Command;
+use byteorder::{ReadBytesExt, WriteBytesExt};
+use bincode::SizeLimit;
+
+const PROTOCOL_VERSION : u8 = 1;
+
+fn main() {
+    // connect to server
+    let stream = TcpStream::connect("127.0.0.1:4242");
+    match stream {
+        Ok(mut s) => {
+            println!("Connected");
+            let rg = receive_greeting(&mut s);
+            if rg == 1 {
+                // close connection
+                println!("Connection closed.");
+                return;
+            }
+
+            // try to send login-data as long as it didn't succeed
+            let mut sl = 1;
+            while sl != 0 {
+                sl = send_login(&mut s);
+            }
+
+            // read commands
+            loop {
+                print!("> ");
+                let e = stdout().flush();
+                match e {
+                    Ok(_) => {},
+                    Err(_) => println!("")
+                }
+                let input = cmd_read();
+
+                // send code for command-package
+                let cs = cmd_send(&mut s, &input);
+                match cs {
+                    0 => return, // end client
+                    _ => continue, // next iteration
+
+                }
+            }
+        } ,
+        Err(e) => panic!("Could not connect to server: {}", e)
+    }
+}
+
+/// Read from command line and return trimmed string
+fn cmd_read() -> String {
+    let mut input = String::new();
+    match io::stdin().read_line(&mut input) {
+        Ok(_) => input.trim().to_string(),
+        _ => "".to_string() //maybe throw PANIC!
+    }
```

Comment:
The `to_string()` is duplicated code here... Luckily Rust is awesome and it's expression based syntax allows this:
```rust
 match io::stdin().read_line(&mut input) {
    Ok(_) => input.trim(),
    _ => "" //maybe throw PANIC!
}.to_string()
```
You may also use `into()` instead of `to_string()` here.

About this function in general: I would call it `read_line`. 
The function `read_line` returning an `Err` value is highly unlikely, so you could just call `unwrap()` in the Result from `read_line`. Or -- a slightly saner way -- you could loop until the `read_line` function does not error (== until you have read a valid String).